src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/ioc/PaymentsService.java

```diff
@@ -115,4 +135,14 @@ private CardPaymentRequest buildPaymentRequest(
             .siteId(siteId)
             .build();
     }
+
+    private Payment from(PaymentDto paymentDto) {
+        return Payment.builder()
+            .amount(paymentDto.getAmount())
+            .reference(paymentDto.getReference())
+            .status(PaymentStatus.fromValue(paymentDto.getStatus()))
+            .dateCreated(paymentDto.getDateCreated().toLocalDate().toString())
+            .nextUrl(paymentDto.getLinks().getNextUrl().getHref().toString())
+            .build();
```

Comment:
Need any NPE protection here?

Reply:
I thought any NPE would bubble up and just make the callback fail, which is what we said we wanted

Reply:
If all of these fields are required then yep; but I wasn't sure that some - actually only `dateCreated` - is really needed, so we might not want to fail if it's not provided by the API.

Reply:
@andrewwa-kainos next url is now the most important one, otherwise we won't be able to redirect the user to progress the payment